src/main/kotlin/org/jetbrains/research/libsl/nodes/references/TypeReference.kt

```diff
@@ -115,3 +115,36 @@ open class TypeReference(
         return result
     }
 }
+
+data class UnionExpressionTypeReference(
+    val left: TypeReference,
+    val right: TypeReference,
+    override val context: LslContextBase
+) : TypeReference(name = "|", context = context)
+
+data class IntersectionExpressionTypeReference(
+    val left: TypeReference,
+    val right: TypeReference,
+    override val context: LslContextBase
+) : TypeReference(name = "&", context = context)
+
+data class LiteralTypeReference(
+    override val name: String,
+    override val context: LslContextBase
+) : TypeReference(name = name, context = context)
```

Comment:
Shouldn't there be a `value` field instead of `name` containing actual/literal value?

Reply:
Fixed;